go/api/base_client.go

```diff
@@ -101,19 +101,33 @@ func (client *baseClient) executeCommand(requestType C.RequestType, args []strin
 		return nil, &ClosingError{"ExecuteCommand failed. The client is closed."}
 	}
 
-	cArgs, argLengths := toCStrings(args)
+	var cArgs []C.uintptr_t
+	var argLengths []C.ulong
+
+	if len(args) > 0 {
+		cArgs, argLengths = toCStrings(args)
+	}
 
 	resultChannel := make(chan payload)
 	resultChannelPtr := uintptr(unsafe.Pointer(&resultChannel))
 
+	// If there are no arguments, pass nil for cArgs and argLengths
```

Comment:
Where do you check for `null`? Just to confirm that we're not dereferencing it.

Reply:
I am checking for null in the executeCommand function, If there are no arguments ( len(cArgs) is 0), I don't dereference the pointers (cArgsPtr and argLengthsPtr ). So they remain nil.